LEAF_Request_Portal/form.php

```diff
@@ -3051,135 +3057,142 @@ public function query($inQuery)
-                $item['approverName'] = $name;
+                require_once 'VAMC_Directory.php';
+                $dir = new VAMC_Directory;
 
-                $data[$item['recordID']]['action_history'][] = $item;
+                $res2 = $this->db->prepared_query('SELECT recordID, stepID, userID, time, description, actionTextPasttense, actionType, comment FROM action_history
+                                                    LEFT JOIN dependencies USING (dependencyID)
+                                                    LEFT JOIN actions USING (actionType)
+                                                    WHERE recordID IN (' . $recordIDs . ')
+                                                    ORDER BY time', array());
+                foreach ($res2 as $item)
+                {
+                    $user = $dir->lookupLogin($item['userID'], true);
+                    $name = isset($user[0]) ? "{$user[0]['Fname']} {$user[0]['Lname']}" : $res[0]['userID'];
+                    $item['approverName'] = $name;
+
+                    $data[$item['recordID']]['action_history'][] = $item;
+                }
             }
-        }
 
-        if($joinRecordResolutionData)
-        {
-            $res2 = $this->db->prepared_query('SELECT recordID, lastStatus, records_step_fulfillment.stepID, fulfillmentTime FROM records
-                    LEFT JOIN records_step_fulfillment USING (recordID)
-                    LEFT JOIN records_workflow_state USING (recordID)
-                    WHERE recordID IN (' . $recordIDs . ')
-                        AND records_workflow_state.stepID IS NULL
-                        AND submitted > 0
-                        AND deleted = 0', array());
-            foreach ($res2 as $item)
+            if($joinRecordResolutionData)
             {
-                if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null
-                    || $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] < $item['fulfillmentTime']) {
-                    $data[$item['recordID']]['recordResolutionData']['lastStatus'] = $item['lastStatus'];
-                    $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] = $item['fulfillmentTime'];
+                $res2 = $this->db->prepared_query('SELECT recordID, lastStatus, records_step_fulfillment.stepID, fulfillmentTime FROM records
+                        LEFT JOIN records_step_fulfillment USING (recordID)
+                        LEFT JOIN records_workflow_state USING (recordID)
+                        WHERE recordID IN (' . $recordIDs . ')
+                            AND records_workflow_state.stepID IS NULL
+                            AND submitted > 0
+                            AND deleted = 0', array());
+                foreach ($res2 as $item)
+                {
+                    if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null
```

Comment:
$dfTime = $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'];
if (empty($dfTime) || $dfTime < $item['fulfillmentTime'] {
   $dfTime = $item['fulfillmentTime'];
   ...
   
Basic rule of thumb, if you use an interpreted variable more than once, you generally want to create a separate variable for it.  Makes it not only "cheaper" but easier to read/change by those who follow on.